### PR TITLE
enable shared libs build

### PR DESCRIPTION
**Pull request by mahf708**
We would like to be able to build the entire eamxx project with shared libs enabled, so making the change here.

There may be other changes needed elsewhere, but for now, I believe this is all that is needed.

Putting this PR together so that you can see how this affects your CI and so on.

Tagging @jeff-cohere @singhbalwinder @odiazib @bartgol

### CODE REVIEW

**Comment by jeff-cohere** [comment]
Codecov is being flaky, so we can either ignore the `auto_test` failure or change the [this field] to `true`.

I don't know what's going on with the PNNL CI process. @singhbalwinder or @jaelynlitz , can you take a look?

**Comment by singhbalwinder** [comment]
I have pinged @jaelynlitz on Slack. Hopefully, she can take a look at it soon. If it is urgent (and has been tested offline), I think it is okay to push it.

**Comment by odiazib** [comment]
> Codecov is being flaky, so we can either ignore the `auto_test` failure or change the [this field](https://github.com/eagles-project/mam4xx/blob/main/.github/workflows/auto_test.yml#L118) to `true`.
> 
> I don't know what's going on with the PNNL CI process. @singhbalwinder or @jaelynlitz , can you take a look?

I noted the same behavior when @tcclevenger pushed a PR.

File: CMakeLists.txt [hunk 1]
```diff
@@ -66,9 +66,6 @@ message(STATUS "Configuring MAM4xx v${MAM4XX_VERSION}")
 # Precision of floating point numbers.
 message(STATUS "Using ${HAERO_PRECISION} precision floating point numbers")
 
-# We build static libraries only.
-set(BUILD_SHARED_LIBS OFF)
```
Comment (odiazib) [inline CMakeLists.txt:70]: @jeff-cohere, @mahf708. Should we add CMake options to turn BUILD_SHARED_LIBS on and off, or can we set BUILD_SHARED_LIBS=ON/OFF directly from the CMake configuration?

**Review by jeff-cohere — APPROVED** [review]

**Pull request merged**